src/BitStringAddresses/bitstring.jl

```diff
@@ -0,0 +1,307 @@
+    top_chunk_bits(s::BitString)
+
+Number of bits stored in top chunk. Equivalent to `chunk_bits(s, 1)`.
+"""
+function top_chunk_bits(::Type{<:BitString{B}}) where B
+    return B % 64 == 0 ? 64 : B % 64
+end
+
+for f in (:num_chunks, :num_bits, :top_chunk_bits)
+    @eval $f(s::BitString) = $f(typeof(s))
+end
+
+"""
+    chunks(::Type{<:BitString})
+    chunks(s)
+
+`SVector` that stores the chunks of `s`.
+"""
+chunks(s::BitString) = s.chunks
+"""
+    chunks_bits(s, i)
+
+Number of bits in the `i`-th chunk of `s`.
+"""
+chunk_bits(s, i) = chunk_bits(typeof(s), i)
+chunk_bits(::Type{<:BitString{B,1}}, _) where {B} = B
+function chunk_bits(::Type{S}, i) where {S<:BitString}
+    return ifelse(i == 1, top_chunk_bits(S), 64)
+end
+
+function ghost_bit_mask(::Type{S}) where S<:BitString
+    unused_bits = 64 - top_chunk_bits(S)
+    return ~zero(UInt64) >>> unused_bits
+end
+
+"""
+    remove_ghost_bits(s)
+
+Remove set bits outside data field if any are present.
+
+See also: [`has_ghost_bits`](@ref).
+"""
+function remove_ghost_bits(s::S) where {S<:BitString}
+    mask = ghost_bit_mask(S)
+    return S(setindex(s.chunks, s.chunks[1] & mask, 1))
+end
+
+@inline function remove_ghost_bits(s::S) where {S<:BitString{<:Any,1}}
+    mask = ghost_bit_mask(S)
+    return S(chunks(s) .& mask)
+end
+
+"""
+    has_ghost_bits(s)
+
+Check for bits outside data field.
+
+See also: [`remove_ghost_bits`](@ref).
+"""
+function has_ghost_bits(s::S) where {S<:BitString}
+    top = first(chunks(s))
+    mask = ~zero(UInt64) << top_chunk_bits(S)
+    return top & mask > 0
+end
+
+###
+### Alternative/useful constructors. These are not super efficient, but they are safe.
+###
+function BitString{B}(i::Union{Int128,Int64,Int32,Int16,Int8}) where {B}
+    return remove_ghost_bits(BitString{B}(unsigned(i)))
+end
+function BitString{B}(i::Union{UInt64,UInt32,UInt16,UInt8}) where {B}
+    N = num_chunks(Val(B))
+    s = setindex(zero(SVector{N,UInt64}), UInt64(i), N)
+    return remove_ghost_bits(BitString{B,N}(s))
+end
+function BitString{B}(i::UInt128) where {B}
+    N = num_chunks(Val(B))
+    left = i >>> 0x40 % UInt64
+    right = i & typemax(UInt64) % UInt64
+    s = ntuple(Val(N)) do i
+        i == N ? right : i == N - 1 ? left : zero(UInt64)
+    end
+    return remove_ghost_bits(BitString{B,N}(SVector{N,UInt64}(s)))
+end
+function BitString{B}(i::BigInt) where {B}
+    N = num_chunks(Val(B))
+    s = zero(SVector{N,UInt64})
+    j = N
+    while i ≠ 0
+        chunk = i & typemax(UInt64) % UInt64
+        i >>>= 64
+        s = setindex(s, chunk, j)
+        j -= 1
+    end
+    return remove_ghost_bits(BitString{B,N}(s))
+end
+
+function Base.zero(S::Type{<:BitString{B}}) where {B}
+    N = num_chunks(Val(B))
+    BitString{B,N}(zero(SVector{N,UInt64}))
+end
+Base.zero(s::BitString) = zero(typeof(s))
+
+function Base.show(io::IO, s::BitString{B,N}) where {B,N}
+    print(io, "BitString{$B,$N}(", join(map(i -> repr(i)[3:end], s.chunks), '|'), ')')
+end
```

Comment:
It would be good to have `show` print a `BitString` in a form that is a constructor as well. Then you could copy and paste the output into the REPL to create a `BitString`.

E.g. `BitString{19,1}(0x0000000000055555)`. 

Currently it prints `BitString{19,1}(0000000000055555)`.

Reply:
Done. Changed it to always show as `BitString{B}(big"0x...")`, so it also works for large bitstrings.